packages/replayer/src/screenshot.utils.ts

```diff
@@ -7,13 +7,13 @@ import { prepareScreenshotsDir } from "./output.utils";
 export interface TakeScreenshotOptions {
   page: Page;
   outputDir: string;
-  screenshotSelector: string;
+  screenshotSelector: string | undefined;
```

Comment:
I changed the cases where we were previously converting to empty string, to instead passing undefined through.

Converting to empty string makes sense for raw javascript where you don't have strict null checks, and so want to reduce the chance of a null pointer exception, and so I've often seen it in JS codebases. But for typescript the tradeoffs are different. When its typed as just string the reader has no hint that the screenshot selector could be absent (represented as ""), and so may not think to handle this case. However when it's `string | undefined` they'll know that it may be absent, and will normally be forced by typescript to handle the absent case. I've found  therefore that the explicit | undefined typings can reduce bugs and aids understanding for other readers.